poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SheetDataWriter.java

```diff
@@ -397,46 +397,51 @@ protected void outputEscapedString(String s) throws IOException {
             return;
         }
 
-        for (Iterator<String> iter = CodepointsUtil.iteratorFor(s); iter.hasNext(); ) {
-            String codepoint = iter.next();
+        for (int i = 0; i < s.length(); i++) {
+            final int codepoint = s.codePointAt(i);
```

Comment:
I might need to improve the readability of the loop, however I think this is covered by line 441, where (in case of a character pair) the counter is additionally increased.

Reply:
loops should not increment in multiple places (ideally) - if you use the codePintCount for the limit of the loop then the count will be correct

Reply:
Yes, but then the codePointAt() method will not receive the correct index. I switched to using the codePoints iterator, which does not seem to have a big performance impact.